src/modules/flight_mode_manager/tasks/Transition/FlightTaskTransition.cpp

```diff
@@ -120,3 +108,123 @@ bool FlightTaskTransition::update()
 	_yaw_setpoint = NAN;
 	return ret;
 }
+
+void FlightTaskTransition::updateParameters()
+{
+	// check for parameter updates
+	if (_parameter_update_sub.updated()) {
+		// clear update
+		parameter_update_s pupdate;
+		_parameter_update_sub.copy(&pupdate);
+
+		updateParametersFromStorage();
+	}
+}
+
+void FlightTaskTransition::updateParametersFromStorage()
+{
+	if (_param_handle_pitch_cruise_degrees != PARAM_INVALID) {
+		param_get(_param_handle_pitch_cruise_degrees, &_param_pitch_cruise_degrees);
+	}
+
+
+	if (_param_handle_vt_b_dec_i != PARAM_INVALID) {
+		param_get(_param_handle_vt_b_dec_i, &_param_vt_b_dec_i);
+	}
+
+
+	if (_param_handle_vt_b_dec_mss != PARAM_INVALID) {
+		param_get(_param_handle_vt_b_dec_mss, &_param_vt_b_dec_mss);
+	}
+
+}
+
+void FlightTaskTransition::updateSubscribers()
+{
+	_sub_vehicle_status.update();
+	_sub_position_sp_triplet.update();
+	_sub_vehicle_local_position.update();
+}
+
+bool FlightTaskTransition::isVtolFrontTransition() const
+{
+	return _sub_vehicle_status.get().in_transition_mode
+	       && _sub_vehicle_status.get().in_transition_to_fw;
+
+}
+
+bool FlightTaskTransition::isVtolBackTransition() const
+{
+	return _sub_vehicle_status.get().in_transition_mode
+	       && !_sub_vehicle_status.get().in_transition_to_fw;
+}
+
+float FlightTaskTransition::computeBackTranstionPitchSetpoint()
+{
+	const vehicle_local_position_s &local_pos = _sub_vehicle_local_position.get();
+	const position_setpoint_s &current_pos_sp = _sub_position_sp_triplet.get().current;
+
+	// Retrieve default decelaration setpoint
+	const float default_deceleration_sp = _param_vt_b_dec_mss;
+	// Maximum allowed deceleration setpoint as a function of the nominal deceleration setpoint
+	const float max_deceleration_sp = 2.f * default_deceleration_sp;
+
+	float deceleration_sp = default_deceleration_sp;
+
+	const float track = atan2f(local_pos.vy, local_pos.vx);
+	const float accel_body_forward = cosf(track) * local_pos.ax + sinf(track) * local_pos.ay;
+	const float vel_body_forward = cosf(track) * local_pos.vx + sinf(track) * local_pos.vy;
+
+
+	if (current_pos_sp.valid && local_pos.xy_valid) {
```

Comment:
As you're using global position and not just local I would check if global is actually valid:
```suggestion
	if (current_pos_sp.valid && local_pos.xy_valid && local_pos.xy_global) {
```